### PR TITLE
Revendor non-suffixed crates which have been updated

### LINKED ISSUE #33: Updating existing vendored crate
In our setup, when we want to upgrade an existing dependency we
1) Run the upgrade (eg futures 0.1.3 => 0.1.13) (get a new Cargo.lock file with the deps)
2) `rm -r vendor/`
3) Run `cargo vendor`

Ideally, it would be possible to eliminate step 2. Currently if `vendor/futures/.cargo-checksum.json` exists, it won't overwrite it. It would be nice if cargo vendor would verify if the crate is the right version.

This issue only comes up if `--explicit-version` is not set. If explicit version is set, then
`vendor/futures-0.1.3/.cargo-checksum.json` would be separate from
`vendor/futures-0.1.13/.cargo-checksum.json`

### PR DESCRIPTION
**Pull request by nipunn1313**
Fixes #33

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -170,16 +172,24 @@ fn sync(lockfile: &Path,
...
+        } else {
+            // Eg vendor/futures
+            id.name().to_string()
         };
         let dst = local_dst.join(&dst_name);
+
         let cksum = dst.join(".cargo-checksum.json");
-        if cksum.exists() {
+        if dir_has_version_suffix && cksum.exists() {
+            // Always re-copy directory without version suffix in case the version changed
```
Comment (nipunn1313) [inline src/main.rs]: Should we open up the Cargo.toml to check if the version has actually changed? That would prevent a lot of FS copying, but I'm not sure it matters too much. Vendoring is not typically on the path that needs to be fast.

In my case, I wouldn't mind unconditionally re-vendoring everything (eg wiping the vendor/ directory) every time. But this at least fixes the bug.

File: src/main.rs [hunk 2]
```diff
@@ -170,16 +172,24 @@ fn sync(lockfile: &Path,
...
             continue
         }
+
+        println!("Vendoring {} to {}", id, dst.display());
```
Comment (nipunn1313) [inline src/main.rs]: I found this line helpful, but I'm not tied to it.

File: src/main.rs (hunk 1 above)
Comment (alexcrichton) [inline src/main.rs]: Nah this is fine, I think this was probably premature optimization on my part which made the tool unnecessarily hard to use!

One use case I'll eventually want to support is to vendor multiple lock files into one directory which may necessitate not blowing away the `vendor` dir, but then again you could also pass multiple arguments...

File: src/main.rs (hunk 2 above)
Comment (alexcrichton) [inline src/main.rs]: 👍

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**